numpy/lib/format.py

```diff
@@ -259,6 +259,43 @@ def dtype_to_descr(dtype):
     else:
         return dtype.str
 
+def descr_without_paddings(descr):
+    aligned = False
+    newdescr = {'names': [], 'formats': [], 'offsets': []}
+    offset = 0
+    unique_name = 'invisible_field_for_internal_descr_without_paddings'
+    for itr, tup in enumerate(descr):
+        if tup[0]=='':
+            if len(tup) == 2:
+                fmt = numpy.dtype(tup[1])
+                if fmt.char == 'V':
+                    #ignore it as padding
+                    offset += fmt.itemsize
+                    continue
+            # replace the empty name with a hopefully unique name
+            # so numpy.dtype(tup) will not fail
+            tup = ('%s%d' % (unique_name, itr),) + tup[1:]
+        if isinstance(tup[1], list):
+            # record of records, recurse
+            if len(tup) > 2:
```

Comment:
Can this ever be 4, or is `(name, type, shape)` the longest possible tuple? What about titles?

Reply:
test, fixed for titles